File: Src/WitsmlExplorer.Api/Jobs/DeleteJobs.cs
using WitsmlExplorer.Api.Jobs.Common;

namespace WitsmlExplorer.Api.Jobs
{
    public record DeleteComponentsJob : IDeleteJob<ComponentReferences> { }
    public record DeleteObjectsJob : IDeleteJob<ObjectReferences> { }
    public record DeleteWellboreJob : IDeleteJob<WellboreReference>
    {
        public bool CascadedDelete { get; init; }
    }
    public record DeleteWellJob : IDeleteJob<WellReference>
    {
        public bool CascadedDelete { get; init; }
    }
}

File: Src/WitsmlExplorer.Frontend/models/jobs/deleteJobs.ts
import ComponentReferences from "models/jobs/componentReferences";
import ObjectReferences from "models/jobs/objectReferences";
import WellboreReference from "models/jobs/wellboreReference";

export interface DeleteObjectsJob {
  toDelete: ObjectReferences;
}

export interface DeleteComponentsJob {
  toDelete: ComponentReferences;
}

export interface DeleteWellboreJob {
  toDelete: WellboreReference;
  cascadedDelete: boolean;
}

export interface DeleteWellJob {
  toDelete: {
    wellUid: string;
    wellName: string;
  };
  cascadedDelete: boolean;
}
